runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -469,4 +488,64 @@ public void onNpcDespawned(NpcDespawned npcDespawned)
 		NPC npc = npcDespawned.getNpc();
 		npcs.remove(npc);
 	}
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired scriptPostFired)
+	{
+		if (scriptPostFired.getScriptId() == ScriptID.AGI_LAPS_CHECK)
+		{
+			Widget agiLapsHeader = client.getWidget(WidgetInfo.AGI_LAPS_HEADER);
+			if (agiLapsHeader != null)
+			{
+				if (agiLapsHeader.getText().equals("Agility Course Lap Counts"))
+				{
+					Widget agiCourseLapCount = client.getWidget(WidgetInfo.AGI_LAPS_COUNT);
+					String[] lapsList;
+					try
+					{
+						lapsList = agiCourseLapCount.getText().split("<br>");
+					}
+					catch (NullPointerException nullPointerException)
+					{
+						return;
+					}
+					List<String[]> parseList = new ArrayList<>(lapsList.length);
```

Comment:
All of this parsing seems extremely excessive when it seems like each line could be matched with a regular expression. `Pattern.compile("(?<course>[\\w\\s]+) - (?<lapCount>\\d+) Laps?")` seems like it would match based on the image I see on Grace's wiki page.